### PR TITLE
Improve our automated resource documentation generation

### PR DESCRIPTION
**Pull request by tas50**
Add more note values to the descriptions, update a few examples for cookstyle and modern gems, and identify a few more resources as providing package on their platform

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/chef/resource/remote_directory.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ class RemoteDirectory < Chef::Resource::Directory
 
       provides :remote_directory
 
-      description "Use the **remote_directory** resource to incrementally transfer a directory from a cookbook to a node. The director that is copied from the cookbook should be located under COOKBOOK_NAME/files/default/REMOTE_DIRECTORY. The remote_directory resource will obey file specificity."
+      description "Use the **remote_directory** resource to incrementally transfer a directory from a cookbook to a node. The director that is copied from the cookbook should be located under `COOKBOOK_NAME/files/default/REMOTE_DIRECTORY`. The `remote_directory` resource will obey file specificity."
```
Comment (phiggins) [inline lib/chef/resource/remote_directory.rb:32]:
```suggestion
      description "Use the **remote_directory** resource to incrementally transfer a directory from a cookbook to a node. The directory that is copied from the cookbook should be located under `COOKBOOK_NAME/files/default/REMOTE_DIRECTORY`. The `remote_directory` resource will obey file specificity."
```

Comment (phiggins) [reply lib/chef/resource/remote_directory.rb:32]: What does "incrementally" mean in this context?

Comment (tas50) [reply lib/chef/resource/remote_directory.rb]: Great question. That's just copied word for word from the docs site.

Comment (phiggins) [reply lib/chef/resource/remote_directory.rb]: I didn't know what "file specificity" meant in this context, and when I looked it up I ended up on a Chef doc page. Maybe worth linking that? https://docs.chef.io/templates/#file-specificity

File: lib/chef/resource/ruby.rb [hunk 2]
```diff
@@ -25,11 +25,7 @@ class Ruby < Chef::Resource::Script
...
-                  " resource may also use any of the actions and properties that are available"\
-                  " to the **execute** resource. Commands that are executed with this resource are (by"\
-                  " their nature) not idempotent, as they are typically unique to the environment"\
-                  " in which they are run. Use `not_if` and `only_if` to guard this resource for idempotence."
+      description "Use the **ruby** resource to execute scripts using the Ruby interpreter. This resource may also use any of the actions and properties that are available to the **execute** resource. Commands that are executed with this resource are (by their nature) not idempotent, as they are typically unique to the environment in which they are run. Use `not_if` and `only_if` to guard this resource for idempotence. Note: The **ruby** script resource (which is based on the **script** resource) is different from the **ruby_block** resource because Ruby code that is run with this resource is created as a temporary file and executed like other script resources, rather than run inline."
```
Comment (phiggins) [inline lib/chef/resource/ruby.rb:28]: > Ruby code that is run with this resource is created as a temporary file and executed like other script resources

This is not true any more.

**Review by phiggins — APPROVED** [review]

File: lib/chef/resource/ruby.rb (hunk 2 above)
Comment (tas50) [reply lib/chef/resource/ruby.rb]: Nuked

**Pull request merged**